diff --git a/system/sepolicy/prebuilts/api/29.0/private/untrusted_app.te b/system/sepolicy/prebuilts/api/29.0/private/untrusted_app.te
old mode 100644
new mode 100755
index c15fa22..68bc5a1
--- a/system/sepolicy/prebuilts/api/29.0/private/untrusted_app.te
+++ b/system/sepolicy/prebuilts/api/29.0/private/untrusted_app.te
@@ -23,3 +23,12 @@ app_domain(untrusted_app)
 untrusted_app_domain(untrusted_app)
 net_domain(untrusted_app)
 bluetooth_domain(untrusted_app)
+
+# Modify by wen.liu for get serial number 2023-7-20
+allow untrusted_app serialno_prop:file { open read map getattr };
+

diff --git a/system/sepolicy/private/untrusted_app.te b/system/sepolicy/private/untrusted_app.te
old mode 100644
new mode 100755
index c15fa22..68bc5a1
--- a/system/sepolicy/private/untrusted_app.te
+++ b/system/sepolicy/private/untrusted_app.te
@@ -23,3 +23,12 @@ app_domain(untrusted_app)
 untrusted_app_domain(untrusted_app)
 net_domain(untrusted_app)
 bluetooth_domain(untrusted_app)
+
+# Modify by wen.liu for get serial number 2023-7-20
+allow untrusted_app serialno_prop:file { open read map getattr };
+

diff --git a/system/sepolicy/public/domain.te b/system/sepolicy/public/domain.te
old mode 100644
new mode 100755
index 32eca92..8eb2e21
--- a/system/sepolicy/public/domain.te
+++ b/system/sepolicy/public/domain.te
@@ -439,7 +439,7 @@ neverallow * *:{ blk_file chr_file } rename;
 
 compatible_property_only(`
     neverallow { domain -init } default_prop:property_service set;
     neverallow { domain -init } mmc_prop:property_service set;
     neverallow { domain -init -vendor_init } exported_default_prop:property_service set;
     neverallow { domain -init } exported_secure_prop:property_service set;
@@ -559,6 +560,8 @@ neverallow {
   -shell
   -system_server
   -vendor_init
+  -untrusted_app # Modify by wen.liu for get serial number 2023-7-20
 } serialno_prop:file r_file_perms;

diff --git a/system/sepolicy/prebuilts/api/29.0/public/domain.te b/system/sepolicy/prebuilts/api/29.0/public/domain.te
old mode 100644
new mode 100755
index 32eca92..8eb2e21
--- a/system/sepolicy/prebuilts/api/29.0/public/domain.te
+++ b/system/sepolicy/prebuilts/api/29.0/public/domain.te
@@ -439,7 +439,7 @@ neverallow * *:{ blk_file chr_file } rename;

compatible_property_only(`
     neverallow { domain -init } default_prop:property_service set;
     neverallow { domain -init } mmc_prop:property_service set;
     neverallow { domain -init -vendor_init } exported_default_prop:property_service set;
     neverallow { domain -init } exported_secure_prop:property_service set;
@@ -559,6 +560,8 @@ neverallow {
   -shell
   -system_server
   -vendor_init
+  -untrusted_app # Modify by wen.liu for get serial number 2023-7-20
 } serialno_prop:file r_file_perms;
